tests/transports/test_all_plugins.py

```diff
@@ -365,278 +297,244 @@ def test_dir_reading_permissions(custom_transport):
         remote_file_name = 'file_remote.txt'
-        retrieved_file_name = os.path.join(local_dir, directory, 'file_retrieved.txt')
+
+        # here use full path in src and dst
+        local_file_abs_path = str(local_dir / directory / local_file_name)
+        retrieved_file_abs_path = str(local_dir / directory / retrieved_file_name)
+        remote_file_abs_path = str(remote_dir / directory / remote_file_name)
 
         text = 'Viva Verdi\n'
-        with open(local_file_name, 'w', encoding='utf8') as fhandle:
+        with open(local_file_abs_path, 'w', encoding='utf8') as fhandle:
             fhandle.write(text)
 
         # localpath is an empty string
         # ValueError because it is not an abs path
         with pytest.raises(ValueError):
-            transport.put('', remote_file_name)
+            transport.put('', remote_file_abs_path)
         with pytest.raises(ValueError):
-            transport.putfile('', remote_file_name)
+            transport.putfile('', remote_file_abs_path)
 
         # remote path is an empty string
         with pytest.raises(OSError):
-            transport.put(local_file_name, '')
+            transport.put(local_file_abs_path, '')
         with pytest.raises(OSError):
-            transport.putfile(local_file_name, '')
+            transport.putfile(local_file_abs_path, '')
 
-        transport.put(local_file_name, remote_file_name)
+        transport.put(local_file_abs_path, remote_file_abs_path)
         # overwrite the remote_file_name
-        transport.putfile(local_file_name, remote_file_name)
+        transport.putfile(local_file_abs_path, remote_file_abs_path)
 
         # remote path is an empty string
         with pytest.raises(OSError):
-            transport.get('', retrieved_file_name)
+            transport.get('', retrieved_file_abs_path)
         with pytest.raises(OSError):
-            transport.getfile('', retrieved_file_name)
+            transport.getfile('', retrieved_file_abs_path)
 
         # local path is an empty string
         # ValueError because it is not an abs path
         with pytest.raises(ValueError):
-            transport.get(remote_file_name, '')
+            transport.get(remote_file_abs_path, '')
         with pytest.raises(ValueError):
-            transport.getfile(remote_file_name, '')
+            transport.getfile(remote_file_abs_path, '')
 
         # TODO : get doesn't retrieve empty files.
         # Is it what we want?
-        transport.get(remote_file_name, retrieved_file_name)
+        transport.get(remote_file_abs_path, retrieved_file_abs_path)
         # overwrite retrieved_file_name
-        transport.getfile(remote_file_name, retrieved_file_name)
-
-        os.remove(local_file_name)
-        transport.remove(remote_file_name)
-        # If it couldn't end the copy, it leaves what he did on
-        # local file
-        assert 'file_retrieved.txt' in transport.listdir('.')
```

Comment:
why this `assert` is removed?

Reply:
Right, I added it back.

Reply:
Thanks! let's put it back before the second `getfile`.
As the first one supposed to already download it, and second one it appears to re-download with `overwrite=True`
 (weird test, but anyways)